test/load_test.c

```diff
@@ -32,168 +38,86 @@ void validate_download_gcg(const DownloadGCGOptions *options, error_code_t expec
+  validate_download_gcg(source_identifier, ERROR_STATUS_SUCCESS);
+}
 
-  validate_download_gcg(&options, ERROR_STATUS_SUCCESS, "Woogles integration test");
+void test_woogles_url_conversion(void) {
+  // Test URL conversion from full Woogles URL
+  const char *source_identifier = "https://woogles.io/game/XuoAntzD";
 
-  config_destroy(config);
-  error_stack_destroy(error_stack);
+  validate_download_gcg(source_identifier, ERROR_STATUS_SUCCESS);
 }
 
-void test_woogles_url_recognition(void) {
-  // Test URL recognition and API call - should succeed in downloading but fail parsing due to no config
-  const char *woogles_url = "https://woogles.io/game/XuoAntzD";
-  
-  DownloadGCGOptions options = {
-    .source_identifier = woogles_url,
-    .lexicon = NULL,
-    .config = NULL  // No config means parsing will fail, but download should work
-  };
+void test_woogles_game_id(void) {
+  // Test URL construction from Woogles alphanumeric identifier
+  const char *source_identifier = "XuoAntzD";
+
+  validate_download_gcg(source_identifier, ERROR_STATUS_SUCCESS);
 
-  // This should fail with "missing config" error, proving download worked
-  validate_download_gcg(&options, ERROR_STATUS_CONFIG_LOAD_MISSING_ARG, "Woogles URL recognition test");
 }
 
 void test_local_file_loading(void) {
-  // Test loading a local GCG file - should succeed in reading but fail parsing due to no config
-  DownloadGCGOptions options = {
-    .source_identifier = "/tmp/test_game.gcg",
-    .lexicon = NULL,
-    .config = NULL  // No config means parsing will fail, but file reading should work
-  };
+  // Test loading from local file (that exists) - use testdata file with fewer moves
+  const char *source_identifier = "testdata/gcgs/success_six_pass.gcg";
 
-  // This should fail with "missing config" error, proving file read worked
-  validate_download_gcg(&options, ERROR_STATUS_CONFIG_LOAD_MISSING_ARG, "Local file loading test");
+  validate_download_gcg(source_identifier, ERROR_STATUS_SUCCESS);
 }
 
 void test_local_file_not_found(void) {
   // Test handling of non-existent local file
-  DownloadGCGOptions options = {
-    .source_identifier = "/tmp/nonexistent_file.gcg",
-    .lexicon = NULL,
-    .config = NULL
-  };
+  const char *source_identifier = "/tmp/nonexistent_file.gcg";
 
-  validate_download_gcg(&options, ERROR_STATUS_FILEPATH_FILE_NOT_FOUND, "Local file not found test");
+  validate_download_gcg(source_identifier, ERROR_STATUS_FILEPATH_FILE_NOT_FOUND);
 }
 
 void test_load(void) {
   printf("Running load tests...\n");
   
   printf("=== Cross-tables Tests ===\n");
-  test_cross_tables_url_conversion();
-  test_cross_tables_integration();
-  test_cross_tables_game_id();
+  test_xt_url_conversion();
+  test_xt_game_id();
+  test_xt_direct_url();
   printf("Cross-tables tests completed.\n\n");
   
   printf("=== Woogles Tests ===\n");
-  test_woogles_url_recognition();
-  test_woogles_integration();
+  test_woogles_url_conversion();
+  test_woogles_game_id();
```

Comment:
We can simplify even further by just directly calling validate_download_gcg from here and inline the source identifier like:

```
validate_download_gcg("/tmp/nonexistent_file.gcg", ERROR_STATUS_FILEPATH_FILE_NOT_FOUND);
```

for each function call.